cpp/include/cuvs/neighbors/cagra.hpp

```diff
@@ -418,31 +649,157 @@ auto build(raft::resources const& handle,
  * @}
  */
 
+/**
+ * @brief Build the index from the dataset for efficient search.
+ *
+ * The build consist of two steps: build an intermediate knn-graph, and optimize it to
+ * create the final graph. The index_params struct controls the node degree of these
+ * graphs.
+ *
+ * It is required that dataset and the optimized graph fit the GPU memory.
+ *
+ * To customize the parameters for knn-graph building and pruning, and to reuse the
+ * intermediate results, you could build the index in two steps using
+ * [cagra::build_knn_graph](#cagra::build_knn_graph) and [cagra::optimize](#cagra::optimize).
+ *
+ * The following distance metrics are supported:
+ * - L2
+ *
+ * @param[in] res
+ * @param[in] params parameters for building the index
+ * @param[in] dataset a matrix view (device) to a row-major matrix [n_rows, dim]
+ * @param[out] idx the constructed cagra index
+ */
 void build_device(raft::resources const& handle,
```

Comment:
Oversight- should be removed from the public APIs (note that these were copied directly from RAFT)

Reply:
If you have a moment, it would be helpful to remove build_host and build_device. That was an oversight. Those are invoked internally to `build()` but shouldn't be part of the public APIs. It also creates confusion for users as to what APIs are available for them. `hpp` files should only include APIs we expect users to interact with.

Reply:
removed.